src/client.js

```diff
@@ -18,18 +18,18 @@ const routes = getRoutes(store);
 const devComponent = renderDevtools();
 
 // There is probably no need to be asynchronous here
-createRootComponentForClient(store, { routes, history })
+createRootComponentForClient(store, { routes, history }, __PROVIDERS__)
   .then(({ root }) => {
     ReactDOM.render(root, dest);
 
     if (process.env.NODE_ENV !== 'production') {
       window.React = React; // enable debugger
       if (!dest || !dest.firstChild || !dest.firstChild.attributes || !dest.firstChild.attributes['data-react-checksum']) {
-        throw new Error('Server-side React render was discarded. Make sure that your initial render does not contain any client-side code.');
+        console.warn('WARNING: Server-side React render was discarded. Make sure that your initial render does not contain any client-side code.');
```

Comment:
Also unrelated, but with the change to catch errors with the `rootComponent` change this warning was being treated like an error, complete with red herring stack trace.